src/App.tsx

```diff
@@ -14,11 +16,132 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isReverse: boolean;
+  isSortedByAlph: boolean;
+  isSortedByLeng: boolean;
+  selectNumber: number;
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    isReverse: false,
+    isSortedByAlph: false,
+    isSortedByLeng: false,
+    selectNumber: 1,
+  };
+
+  handleButtonReverse = () => {
+    this.setState(state => ({
+      isReverse: !state.isReverse,
+    }));
+  };
+
+  handleButtonReset = () => {
+    this.setState({
+      isReverse: false,
+      isSortedByAlph: false,
+      isSortedByLeng: false,
+      selectNumber: 1,
+    });
+  };
+
+  handleButtonSort = () => {
+    this.setState(({ isSortedByAlph }) => ({
+      isSortedByAlph: !isSortedByAlph,
+      isSortedByLeng: false,
+    }));
+  };
+
+  handleButtonSortByLength = () => {
+    this.setState(({ isSortedByLeng }) => ({
+      isSortedByLeng: !isSortedByLeng,
+      isSortedByAlph: false,
+    }));
+  };
+
+  render() {
+    const {
+      isReverse,
+      isSortedByAlph,
+      isSortedByLeng,
+      selectNumber,
+    } = this.state;
+    const goods = [...goodsFromServer].filter(
+      item => item.length >= selectNumber,
+    );
+    const arrForSelect = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+    const buttonClass = 'button is-primary';
+
+    goods.filter(item => item.length >= selectNumber);
+
+    if (isSortedByAlph) {
+      goods.sort((g1, g2) => g1.localeCompare(g2));
+    }
+
+    if (isSortedByLeng) {
+      goods.sort((g1, g2) => g1.length - g2.length);
+    }
+
+    if (isReverse) {
+      goods.reverse();
+    }
+
+    return (
+      <div className="App">
+        <h1 className="title is-1 mt-5">Goods</h1>
+
+        <div className="buttons__container">
+          <button
+            type="button"
+            className={this.state.isReverse
+              ? `${buttonClass} is-active`
+              : `${buttonClass}`}
```

Comment:
you can use classnames lib for conditional classes
https://www.npmjs.com/package/classnames